src/debugger/configuration/resolvers/ros1/launch.ts

```diff
@@ -32,7 +34,31 @@ export class LaunchResolver implements vscode.DebugConfigurationProvider {
         if (!path.isAbsolute(config.target) || path.extname(config.target) !== ".launch") {
             throw new Error("Launch request requires an absolute path as target.");
         }
+
+        const delay = ms => new Promise(res => setTimeout(res, ms));
+
+        switch(env.ROS_VERSION.trim()) {
+            case "1": {
+                // Manage the status of the ROS core, starting one if not present
+                // The ROS core will continue to run until the VSCode window is closed
+                const core_active = rosApi.getCoreStatus();
+                if (!(await core_active).valueOf()) {
+                    rosApi.startCore();
         
+                    // Wait for the core to start
+                    while (!(await rosApi.getCoreStatus()).valueOf()) {
```

Comment:
It would be great to add a maximum timeout - Wouldn't want to cause issues with vscode if roscore fails for some other reason.

Reply:
Good catch. I wasn't sure on the exact timeout to set, so I settled for 3 seconds. From my local testing, roscore seems to activate at about the 700 ms mark, but I figured it could be slower if you were running this on a Raspberry Pi or something. The error message should help users if a super slow machine can't start roscore in 3 seconds.